src/CoreWCF.Queue/src/CoreWCF/Queue/Common/QueueMessageContext.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO.Pipelines;
+using System.Threading;
+using System.Threading.Tasks;
+using CoreWCF.Channels;
+using CoreWCF.Runtime;
+
+namespace CoreWCF.Queue.Common
+{
+    public class QueueMessageContext : RequestContext
+    {
+        public PipeReader QueueMessageReader { get; set; }
+        public IDictionary<string, object> Properties { get; set; }
+        private Message _requestMessage;
```

Comment:
I'm a little concerned about extra unnecessary allocation here. I think we should make this virtual, get only, and default to lazily creating a Dictionary here instead of the code which constructs an instance.